packages/mermaid/src/diagrams/sankey/sankeyUtils.ts

```diff
@@ -0,0 +1,29 @@
+const prepareTextForParsing = (text: string): string => {
+  const textToParse = text
+    .replaceAll(/^[^\S\n\r]+|[^\S\n\r]+$/g, '') // remove all trailing spaces for each row
+    .replaceAll(/([\n\r])+/g, '\n') // remove empty lines duplicated
+    .trim();
+
+  return textToParse;
+};
+
+class Uid {
+  private static count = 0;
+  id: string;
+  href: string;
+
+  public static next(name: string): Uid {
+    return new Uid(name + ++Uid.count);
+  }
+
+  constructor(id: string) {
+    this.id = id;
+    this.href = `#${id}`;
+  }
+
+  toString(): string {
+    return 'url(' + this.href + ')';
+  }
+}
```

Comment:
This can be in a more general utils file, others could also use it. 
How are other diagrams handling this now?

Reply:
Like that:
* [C4](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/diagrams/c4/svgDraw.js#L40)
* [Class](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/diagrams/class/svgDraw.js#L43-L72)
* and (skdum tssssss!) [todo "Understand it better" from ER](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/diagrams/er/erRenderer.js#L451-L462)

Reply:
In my opinion, this could be quite a rework, so let it be here for a while, we have to handle other diagrams separately, and make this module common a bit later, too prone to error

Reply:
No need to modify other diagrams now, simply moving this to a common file will enable future diagrams to use it.

Reply:
Put Uid to separate file, we may move it somewhere else if needed